scylla-gdb.py

```diff
@@ -4972,6 +4973,113 @@ def invoke(self, arg, from_tty):
             gdb.write("[{}] 0x{:x} {}\n".format(offset + i, obj, sym_text))
 
 
+class scylla_large_objects(gdb.Command):
+    """List live objects from one of the seastar allocator's large pools
+
+    The pool is selected with the `-o|--object-size` flag. Results are paginated by
+    default as there can be thousands of objects. Default page size is 20.
+    To list a certain page, use the `-p|--page` flag. To find out the number of
+    total objects and pages, use `--summarize`.
+    To sample random pages, use `--random-page`.
+
+    For usage see: scylla large-objects --help
+
+    Examples:
+
+    (gdb) scylla large-objects -o 32768 --summarize
+    number of objects: 9737
+    page size        : 20
+    number of pages  : 487
+
+    (gdb) scylla large-objects -o 131072 --random-page
+    [40] 0x60001cb40000
+    [41] 0x60001cb60000
+    [42] 0x60001cb80000
+    [43] 0x60001cba0000
+    [44] 0x60001cbc0000
+    [45] 0x60001cbe0000
+    [46] 0x60001cc00000
+    [47] 0x60001cc20000
+    [48] 0x60001cc40000
+    [49] 0x60001cc60000
+    [50] 0x60001cc80000
+    [51] 0x60001cca0000
+    [52] 0x60001ccc0000
+    [53] 0x60001cce0000
+    [54] 0x60001cd00000
+    [55] 0x60001cd20000
+    [56] 0x60001cd40000
+    [57] 0x60001cd60000
+    [58] 0x60001cd80000
+    [59] 0x60001cda0000
+    """
+
+    def __init__(self):
+        gdb.Command.__init__(self, 'scylla large-objects', gdb.COMMAND_USER, gdb.COMPLETE_COMMAND)
+        self._parser = None
+        self._objects = None
+        self._object_size = None
+
+    def init_parser(self):
+        parser = argparse.ArgumentParser(description="scylla large-objects")
+        parser.add_argument("-o", "--object-size", action="store", type=int, required=True,
+                help="Object size, valid sizes are: ")
```

Comment:
unlike small objects, we don't have small object pool. so this help message might not apply anymore. probably could  put

```py
max_small_allocation = scylla_small_objects.get_object_sizes()[-1]
# ...
help=f"Object size greater than {max_small_allocation}"
```

Reply:
Done in v2.